src/node_file-inl.h

```diff
@@ -156,8 +156,10 @@ FSReqPromise<AliasedBufferT>::New(BindingData* binding_data,
 
 template <typename AliasedBufferT>
 FSReqPromise<AliasedBufferT>::~FSReqPromise() {
-  // Validate that the promise was explicitly resolved or rejected.
-  CHECK(finished_);
+  // Validate that the promise was explicitly resolved or rejected but only if
+  // the Isolate is not terminating because in this case the promise might have
+  // not finished.
+  if (env()->can_call_into_js()) CHECK(finished_);
```

Comment:
`can_call_into_js()` should be used for situations in which we really want to know whether we are able to call into JS, rather than just a generic check whether the Environment is being torn down.

```suggestion
  if (env()->is_stopping()) CHECK(finished_);
```